src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -116,6 +116,10 @@ INamedTypeSymbol typeSymbol
 				return;
 			}
 
+			if( Attributes.Objects.ConditionallyImmutable.IsDefined( typeSymbol ) ) {
+				immutabilityContext = immutabilityContext.WithConditionalTypeParametersAsImmutable( typeSymbol );

```

Comment:
Might be some problem here with say...

```csharp
[ConditionallyImmutable]
public sealed class Foo<[ConditionallyImmutable.OnlyIf] T> {

  [ConditionallyImmutable]
  public sealed calss Bar<[ConditionallyImmutable.OnlyIf] U> {
    // something in here....
  }
}
```

But I haven't thought about that much.

Reply:
Open an issue to add test cases?

Reply:
https://github.com/Brightspace/D2L.CodeStyle/issues/670